### PR TITLE
Split e2e CI groups

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING REVIEW

### What does this PR accomplish?
- Splits our gigantic monolithic e2e CI job into smaller, **deterministic** groups
- Cuts down total CI running time:
    - about 50-60% in pessimistic scenario
    - up to 75% in optimistic scenario

[Example of the successful run](https://app.circleci.com/pipelines/github/metabase/metabase/15814/workflows/dab6b411-8f86-49ba-a7ba-d81bf569a1b0) (Total run time: 13m 12s)
[Example of the rerun](https://app.circleci.com/pipelines/github/metabase/metabase/15822/workflows/08b40165-d028-497d-9970-3f10df1e00a4) (Total time: 5m 48s)

### Additional notes:
- Group names (or their size/contents) are not set in stone. We're trying things out. It is quite possible some further rearranging (refinement) will occur in the future. The goal of this PR was to take the first step in that direction.
- It is impossible to make a clear cut between different features / aspects of the app
- Grouping loosely follows the official Metabase documentation structure (but **very** loosely)

_I am sorry in advance for possible merge conflicts due to a lot of files being moved around!_

### Next steps:
- `mongo`, `mysql` and `postgres` e2e groups should not run separately.
- Integrate them into the main groups and make all these DBs available to any test at any given point in time
- Make DB snapshots for each DB and reuse them in tests
    - I spent the last week trying to debug some weird failures and behavior regarding this. The process should be straight-forward but there are still some issues blocking this step from currently going forward.

### CODE REVIEW

**Review by ariya — APPROVED** [review]
Excellent work :1st_place_medal:

File: .circleci/config.yml [hunk 1]
```diff
@@ -895,7 +895,7 @@ jobs:
                       <<: *CacheKeyUberjar
                   - steps: << parameters.before-steps >>
                 command: |
-                  run test-cypress-no-build <<# parameters.only-single-database >> --folder << parameters.test-files-location >> <</ parameters.only-single-database >>
+                  run test-cypress-no-build <<# parameters.test-files >> --spec << parameters.test-files >> <</ parameters.test-files >> --folder << parameters.source-folder >>
```
Comment (flamber) [inline .circleci/config.yml:898]: Just wanna make sure. There should always be `--folder`, while `--spec` should only be there if specific files are called?

**Review by flamber — APPROVED** [review]

**Comment by flamber** [comment]
The negative side-effect of this is that it doesn't generate JARs - I guess we have to enable something on Actions, so it has a JAR artifact available for download on each build.

File: .circleci/config.yml (hunk 1 above)
Comment (nemanjaglumac) [reply .circleci/config.yml:898]: You can use either.
Wnt for the `--folder` flag as the default one. My original idea was to use `--spec` as the default one to be able to arbitrarily combine files from different folders and group them together. Ended up reorganizing files/folders, bu there may still be the need for multiple `--spec` arguments in the future.

You can make a group called `permissions` and pass it a glob for `test-files`:
`/frontend/test/metabase/scenarios/**/permissions.cy.spec.js`

Or you can combine two files from different folders, like in the example from https://github.com/metabase/metabase/pull/15917

**Comment by nemanjaglumac** [comment]
> The negative side-effect of this is that it doesn't generate JARs - I guess we have to enable something on Actions, so it has a JAR artifact available for download on each build.

@flamber unless I misunderstood what you're saying, this has nothing to do with this PR. JARs are **intentionally** not being built when the only files changed are TEST files. Such runs are using cached JARs.
If any source (js, clj or cljs) file is changed, jar is being built again.

This was the tweak that's implemented in https://github.com/metabase/metabase/pull/15900

File: .circleci/config.yml (hunk 1 above)
Comment (flamber) [reply .circleci/config.yml:898]: Okay, so they cannot be used together - it's either or?
Then you wanna do this:
```suggestion
                  run test-cypress-no-build <<# parameters.test-files >> --spec << parameters.test-files >> <</ parameters.test-files >> <<# parameters.source-folder >> --folder << parameters.source-folder >> <</ parameters.source-folder >>
```

Comment (nemanjaglumac) [reply .circleci/config.yml:898]: They can even be used together.
Folder points to the source and specs can still select individual files only from that folder.

**Pull request merged**